src/java.base/linux/classes/sun/nio/fs/LinuxFileSystem.java

```diff
@@ -67,6 +71,115 @@ public Set<String> supportedFileAttributeViews() {
         return SupportedFileFileAttributeViewsHolder.supportedFileAttributeViews;
     }
 
+    private static UnixException catEx(UnixException x, UnixException y) {
+        assert x != null || y != null;
+        UnixException ue = y;
+        if (x != null) {
+            ue = x;
+            if (y != null) {
+                ue.addSuppressed(y);
+            }
+        }
+        return ue;
+    }
+
+    /**
+     * Clones the file whose path name is {@code src} to that whose path
+     * name is {@code dst} using the {@code ioctl} system call with the
+     * {@code FICLONE} request code.
+     *
+     * @param src the path of the source file
+     * @param dst the path of the destination file (clone)
+     * @param followLinks whether to follow links
+     *
+     * @return 0 on success, IOStatus.UNSUPPORTED_CASE if the call does not work
+     *         with the given parameters, or IOStatus.UNSUPPORTED if cloning is
+     *         not supported on this platform
+     */
+    protected int clone(UnixPath src, UnixPath dst, boolean followLinks)
+        throws IOException {
```

Comment:
Formatting nit here too.

Reply:
Fixed in 2157a1480229fa697c824ae814f11a309627d768.